android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginViewModel.kt

```diff
@@ -59,13 +60,22 @@ constructor(
   val launchDialPad
     get() = _launchDialPad
 
+  // -> user-resp (failure)
+  //    -> show-error
+  // -> user-resp (success)
+  //    -> store user info
+  //    -> goto home
   val responseBodyHandler =
     object : ResponseHandler<ResponseBody> {
       override fun handleResponse(call: Call<ResponseBody>, response: Response<ResponseBody>) {
-        response.body()?.run {
-          storeUserPreferences(this)
-          _navigateToHome.value = true
-          _showProgressBar.postValue(false)
+        if (response.isSuccessful)
+          response.body()!!.run {
+            storeUserPreferences(this)
+            _showProgressBar.postValue(false)
+            _navigateToHome.value = true
+          }
+        else {
+          handleFailure(call, RuntimeException("Network call failed with $response"))
```

Comment:
Is it possible to use an exception that will not result into app crash? Unless the `RuntTimeException` is handle I think it will result into an app crash. An `IOException` or it's subclass sounds ideal

Reply:
It does not throw Exception. It uses handleFailure of retrofit -> rightbelow this success handler